lib/rules/padding-line-between-statements.js

```diff
@@ -0,0 +1,585 @@
+    "cjs-export": {
+        test: (node, sourceCode) =>
+            node.type === "ExpressionStatement" &&
+            node.expression.type === "AssignmentExpression" &&
+            CJS_EXPORT.test(sourceCode.getText(node.expression.left))
+    },
+    "cjs-import": {
+        test: (node, sourceCode) =>
+            node.type === "VariableDeclaration" &&
+            node.declarations.length > 0 &&
+            Boolean(node.declarations[0].init) &&
+            CJS_IMPORT.test(sourceCode.getText(node.declarations[0].init))
+    },
+    directive: {
+        test: isDirectivePrologue
+    },
+    expression: {
+        test: (node, sourceCode) =>
+            node.type === "ExpressionStatement" &&
+            !isDirectivePrologue(node, sourceCode)
+    },
+    "multiline-block-like": {
+        test: (node, sourceCode) =>
+            node.loc.start.line !== node.loc.end.line &&
+            isBlockLikeStatement(sourceCode, node)
+    },
+
+    block: newNodeTypeTester("BlockStatement"),
+    empty: newNodeTypeTester("EmptyStatement"),
+
+    break: newKeywordTester("break"),
+    case: newKeywordTester("case"),
+    class: newKeywordTester("class"),
+    const: newKeywordTester("const"),
+    continue: newKeywordTester("continue"),
+    debugger: newKeywordTester("debugger"),
+    default: newKeywordTester("default"),
+    do: newKeywordTester("do"),
+    export: newKeywordTester("export"),
+    for: newKeywordTester("for"),
+    function: newKeywordTester("function"),
+    if: newKeywordTester("if"),
+    import: newKeywordTester("import"),
+    let: newKeywordTester("let"),
+    return: newKeywordTester("return"),
+    switch: newKeywordTester("switch"),
+    throw: newKeywordTester("throw"),
+    try: newKeywordTester("try"),
+    var: newKeywordTester("var"),
+    while: newKeywordTester("while"),
+    with: newKeywordTester("with")
+};
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "require or disallow padding lines between statements",
+            category: "Stylistic Issues",
+            recommended: false
+        },
+        fixable: "whitespace",
+        schema: {
+            definitions: {
+                paddingType: {
+                    enum: Object.keys(PaddingTypes)
+                },
+                statementType: {
+                    anyOf: [
+                        { enum: Object.keys(StatementTypes) },
+                        {
+                            type: "array",
+                            items: { enum: Object.keys(StatementTypes) },
+                            minItems: 1,
+                            uniqueItems: true,
+                            additionalItems: false
+                        }
+                    ]
+                }
+            },
+            type: "array",
```

Comment:
I think it might be better to have each configuration be a different option:

```json
{
  "padding-line-between-statements": [
    "error",
    { "prev": "*", "next": "*", "blankline": "always" },
    { "prev": "return", "next": "*", "blankline": "never" }
  ]
}
```

Then it would be consistent with other rules, such as `no-restricted-syntax`, `no-restricted-properties`, and `no-restricted-globals`. Also, if we ever create a way to extend a shareable config's configuration for a rule without overwriting it, it will be easier if our rules have consistent APIs.